File: frontend/src/types/order.ts
export interface CapturedOrder {
  id: string;
  tenant_id: string;
  driver_id: string;
  device_id: string | null;
  platform: string;
  order_ref: string | null;
  status: string;
  parsed_data: Record<string, unknown> | null;
  amount: number | null;
  captured_at: string;
}

export interface OrderFilters {
  date_from?: string;
  date_to?: string;
  driver_id?: string;
  platform?: string;
  page?: number;
  per_page?: number;
}

export interface OrderSummary {
  date: string;
  total: number;
  by_platform: Record<string, number>;
  total_amount: number;
  by_platform_amount: Record<string, number>;
  top_drivers: { driver_id: string; driver_name: string; count: number }[];
}

export interface HourlyDistribution {
  hour: number;
  count: number;
}

File: backend/app/schemas/order.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class CapturedOrderResponse(BaseModel):
    id: uuid.UUID
    tenant_id: uuid.UUID
    driver_id: uuid.UUID
    device_id: uuid.UUID | None
    platform: str
    order_ref: str | None
    status: str
    parsed_data: dict | None
    amount: Decimal | None
    captured_at: datetime

    model_config = {"from_attributes": True}


class OrderSummaryResponse(BaseModel):
    date: str
    total: int
    by_platform: dict[str, int]
    total_amount: float
    by_platform_amount: dict[str, float]
    top_drivers: list[dict]

class HourlyDistribution(BaseModel):
    hour: int
    count: int
